docs/src/types/bvp_types.md

```diff
@@ -47,9 +47,16 @@ bc!(residual, sol)
 
 where `u` is the current solution to the ODE which is used to compute the `residual`.
 Note that all features of the `ODESolution` are present in this form.
-In both cases, the size of the residual matches the size of the initial condition
-(for more general problems, use the
-[parameter estimation routines](../../analysis/parameter_estimation.html)).
+In both cases, the size of the residual matches the size of the initial condition.
+For more general problems, use the
+[parameter estimation routines](../../analysis/parameter_estimation.html) and change the signature of `bc!` and `f` to
+
+```julia
+bc!(residual, sol, p)
+f(t, u, p)
```

Comment:
should mention the pre-allocated output here, `f(t,u,p,du)`.

Reply:
added it